### PR TITLE
Simplify dial to use atan2

### PR DESCRIPTION
**Pull request by NigelBreslaw**
Suspect some of the simplification could have been done with the original. But this wasn't so obvious at the time when the math was a bit more fussy.

### CODE REVIEW

File: examples/dial/dial.slint [hunk 1]
```diff
@@ -11,11 +11,7 @@ export global AppState {
     in-out property <length> elementRadius: 185px;
 
     pure public function normalizeAngle(angle: angle) -> angle {
-        if angle >= 0 {
-            return Math.mod(angle, 360deg);
-        } else {
-            return 360deg + Math.mod(angle, 360deg);
-        }
+        return (angle + 360deg).mod(360deg);
```
Comment (tronical) [inline examples/dial/dial.slint:14]:
```suggestion
        return (angle + 360deg).mod(360deg);
```

Isn't this a no-op? :)

Comment (NigelBreslaw) [reply examples/dial/dial.slint:14]: Knowing me, possibly. However atan2 returns an angle between -180 to 180. Then unlike awesome languages such as javascript, mod in slint can return a negative number. So the +360deg ensures its a positive number. Then its also used to normalise the first value when you put your finger on the dial. Say the dial is already at 45 degrees the dial records that and subtracts the value of where you places your finger on the dial. Again possibly creating a negative value. So this ensure everything is always between 0 and 359. Did I get it right?

Comment (tronical) [reply examples/dial/dial.slint:14]: Thank you, this is a crucial detail I was missing. You’re right:)

Comment (ogoffart) [reply examples/dial/dial.slint:14]: honestly, I think we may consider changing this behavior (of mod)
I can't think of any practical reason to return a negative number

**Review by ogoffart — APPROVED** [review]

File: examples/dial/dial.slint [hunk 2]
```diff
@@ -91,30 +88,14 @@ export component AppWindow inherits Window {
                     moved => {
                         relativeX = ta.mouse-x - centerX;
                         relativeY = ta.mouse-y - centerY;
```
[the remark this answers is not in the document]
Comment (NigelBreslaw) [reply examples/dial/dial.slint:90]: They need to be updated every time `moved` is triggered?

**Pull request merged**